js/js.libraries/src/core/collections.kt

```diff
@@ -85,13 +85,34 @@ public fun <K, V> mapOf(pair: Pair<K, V>): Map<K, V> = hashMapOf(pair)
  * Each element in the list gets replaced with the [value].
  */
 @SinceKotlin("1.2")
-public fun <T> MutableList<T>.fill(value: T) {
-    for (index in 0..lastIndex) {
+public fun <T> MutableList<T>.fill(value: T): Unit {
+    for (index in indices) {
         this[index] = value
     }
 }
 
 /**
+ * Randomly shuffles elements in this list.
+ *
+ * See: https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm
+ */
+@SinceKotlin("1.2")
+public fun <T> MutableList<T>.shuffle(): Unit {
+    (lastIndex downTo 1).forEach { i ->
```

Comment:
JS backend doesn't optimize such loops, so it would be better to rewrite it as plain old `while` loop.

Reply:
Sure, thanks!

Reply:
`for (i in lastIndex downTo 1) { ... }` is already supported by optimizer

Reply:
Thanks @konsoletyper, that's exactly what I changed it to :)